server/channels/app/platform/service.go

```diff
@@ -323,11 +323,10 @@ func New(sc ServiceConfig, options ...Option) (*PlatformService, error) {
 	}
 
 	ps.AddLicenseListener(func(oldLicense, newLicense *model.License) {
-		if (oldLicense == nil && newLicense == nil) || !ps.startMetrics {
-			return
-		}
+		wasLicensed := (oldLicense != nil && *oldLicense.Features.Metrics) || (model.BuildNumber == "dev")
+		isLicensed := (newLicense != nil && *newLicense.Features.Metrics) || (model.BuildNumber == "dev")
 
-		if oldLicense != nil && newLicense != nil && *oldLicense.Features.Metrics == *newLicense.Features.Metrics {
+		if wasLicensed == isLicensed || !ps.startMetrics {
```

Comment:
This is a mind bender. Are we sure this is correct? It appears to call `RestartMetrics` if the license state has changed, but there appears to be no actual license check here? Is it handled somewhere else?

Reply:
Agree it's convoluted. We do the actual check in 

https://github.com/mattermost/mattermost/blob/8a08714ae15c1361f1a0cb3deab4b327be2d5ead/server/enterprise/metrics/metrics.go#L954-L962